parachain/pallets/inbound-queue/src/lib.rs

```diff
@@ -322,9 +327,14 @@ pub mod pallet {
 	/// API for accessing the delivery cost of a message
 	impl<T: Config> Get<BalanceOf<T>> for Pallet<T> {
 		fn get() -> BalanceOf<T> {
-			let pricing_parameters = T::PricingParameters::get();
-			let refund: BalanceOf<T> = T::WeightToFee::weight_to_fee(&T::WeightInfo::submit());
-			refund.saturating_add(pricing_parameters.rewards.local)
+			let weight_fee = T::WeightToFee::weight_to_fee(&T::WeightInfo::submit());
+			let length_fee = T::LengthToFee::weight_to_fee(&Weight::from_parts(
+				T::MaxMessagePayloadSize::get() as u64,
```

Comment:
I looked at this problem earlier today. We need to consider the maximum possible length of the `message` parameter passed to submit. According to my rough estimates of receipt proof length, and transaction receipt size, this is going to be around 2KiB.